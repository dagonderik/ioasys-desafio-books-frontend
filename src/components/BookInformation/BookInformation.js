import React from 'react';

const BookInformation = ({route, bookList}) => {
    return (
        <div>
            <img alt='' src={`${bookList[route].imageUrl}`} style={{display:"inline-block", minWidth:"30vh", maxWidth:"50vh", height:"10%", align: "center",verticalAlign: "middle"}} />
        
            <div style={{display:"inline-block", verticalAlign: "middle", marginLeft: "10px"}}>
            <h1 className='black f1'>{bookList[route].title}</h1>
            <p className='hot-pink f4'>
                {bookList[route].authors.map((user,i) => {
                return( <>{bookList[route].authors[i]}  </>)
                })}
            </p>
            <div>
                <p  className='f4 b black'>INFORMAÇÕES<br/></p>
                <p style={{display:"flex", justifyContent: "space-between"}}><b>Páginas</b> {bookList[route].pageCount}</p>
                <p style={{display:"flex", justifyContent: "space-between"}}><b>Editora</b> {bookList[route].publisher}</p>
                <p style={{display:"flex", justifyContent: "space-between"}}><b>Publicação</b> {bookList[route].published}</p>
                <p style={{display:"flex", justifyContent: "space-between"}}><b>Idioma</b> {bookList[route].language}</p>
                <p style={{display:"flex", justifyContent: "space-between"}}><b>Título Original</b> {bookList[route].title}</p>
                <p style={{display:"flex", justifyContent: "space-between"}}><b>ISBN-10</b> {bookList[route].isbn10}</p>
                <p style={{display:"flex", justifyContent: "space-between"}}><b>ISBN-13</b> {bookList[route].isbn13}</p>
            
            <p className='f4 b black'>RESENHA DA EDITORA</p>
            <p className='f5' style={{position: "absolute"}} >{bookList[route].description}</p>
            </div>
            </div>
        </div>
    );
}

export default BookInformation;