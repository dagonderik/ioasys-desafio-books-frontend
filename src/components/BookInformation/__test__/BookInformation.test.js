import React from "react";
import ReactDOM from 'react-dom';
import BookInformation from "../BookInformation";
import renderer from "react-test-renderer";

const test=[
    {
        id:1,
        authors:["authors"],
        title:"title",
        pageCount:1,
        publisher:"publisher",
        published:"published",
        imageUrl:"imageURL"
    },
    {
        id:2,
        authors:["authors"],
        title:"title",
        pageCount:1,
        publisher:"publisher",
        published:"published",
        imageUrl:"imageURL"
    },
    {
        id:3,
        authors:["authors"],
        title:"title",
        pageCount:1,
        publisher:"publisher",
        published:"published",
        imageUrl:"imageURL"
    }

]

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BookInformation
    route={1}
    bookList={test}></BookInformation>,div)
})

it("matches snapshot", () =>{
    const tree = renderer.create(<BookInformation
        route={1}
        bookList={test}></BookInformation>).toJSON();
    expect(tree).toMatchSnapshot();
})