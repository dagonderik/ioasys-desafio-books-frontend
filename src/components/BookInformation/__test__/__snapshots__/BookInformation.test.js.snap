// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="bookList"
>
  <img
    alt=""
    src="imageURL"
    style={
      Object {
        "align": "center",
        "display": "inline-block",
        "height": "10%",
        "maxWidth": "30vh",
        "verticalAlign": "middle",
      }
    }
  />
  <div
    style={
      Object {
        "display": "inline-block",
        "marginLeft": "10px",
        "verticalAlign": "middle",
      }
    }
  >
    <h1
      className="black f1"
    >
      title
    </h1>
    <p
      className="hot-pink f4"
    >
      authors
        
    </p>
    <div>
      <p
        className="f4 b black"
      >
        INFORMAÇÕES
        <br />
      </p>
      <p
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <b>
          Páginas
        </b>
         
        1
      </p>
      <p
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <b>
          Editora
        </b>
         
        publisher
      </p>
      <p
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <b>
          Publicação
        </b>
         
        published
      </p>
      <p
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <b>
          Idioma
        </b>
         
      </p>
      <p
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <b>
          Título Original
        </b>
         
        title
      </p>
      <p
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <b>
          ISBN-10
        </b>
         
      </p>
      <p
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <b>
          ISBN-13
        </b>
         
      </p>
      <p
        className="f4 b black"
      >
        RESENHA DA EDITORA
      </p>
      <p
        className="f5"
        style={
          Object {
            "position": "absolute",
          }
        }
      />
    </div>
  </div>
</div>
`;
